File: selenium/pages/loginPage.js
const { BasePage } = require("./basePage");

class LoginPage extends BasePage {
  async assertLoaded() {
    await this.assertHasText("SMART FURNACE AI");
    await this.assertHasText("ACCESS TERMINAL");
  }

  async enterEmail(email) {
    await this.type("email", email);
  }

  async enterPassword(password) {
    await this.type("password", password);
  }

  async submit() {
    await this.tapText("ACCESS TERMINAL");
  }

  async goToRegister() {
    await this.tapText("No access? Request an account here.");
  }

  async login(email, password) {
    await this.enterEmail(email);
    await this.enterPassword(password);
    await this.submit();
  }
}

module.exports = { LoginPage };

File: appium/pages/login_page.py
from pages.base_page import BasePage


class LoginPage(BasePage):
    def assert_loaded(self) -> None:
        assert self.has_text("SMART FURNACE AI")
        assert self.has_text("ACCESS TERMINAL")

    def enter_email(self, email: str) -> None:
        self.type_text("email", email, fallback_label="Institutional Email")

    def enter_password(self, password: str) -> None:
        self.type_text("password", password, fallback_label="Password")

    def submit(self) -> None:
        self.tap_text("ACCESS TERMINAL")

    def go_to_register(self) -> None:
        self.tap_text("No access? Request an account here.")

    def login(self, email: str, password: str) -> None:
        self.enter_email(email)
        self.enter_password(password)
        self.submit()
